pulumitest/writePulumiYAML.go

```diff
@@ -0,0 +1,20 @@
+package pulumitest
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func (a *PulumiTest) WritePulumiYaml(t PT, program string) {
+	t.Helper()
+
+	// YAML doesn't allow tabs but go uses tabs which makes for a miserable experience with inline yaml programs
+	program = strings.ReplaceAll(program, "\t", "    ")
```

Comment:
Let's validate there's no tabs instead of attempting to auto-fix because this might end up being invalidly indented. We should throw an error which should indicate where the tab characters are and could suggest the fixed indenting within the error message.